ipie/estimators/utils.py

```diff
@@ -136,4 +136,7 @@ def gab_spin(A, B, na, nb):
     GA, GAH = gab_mod(A[:, :na], B[:, :na])
     if nb > 0:
         GB, GBH = gab_mod(A[:, na:], B[:, na:])
+    else:
+        GB = numpy.zeros(GA.shape)
+        GBH = numpy.zeros((0, GAH.shape[1]))
```

Comment:
add a dtype for GB and GBH here using GA's dtype, just to avoid any complex casting errors down the line.